File: responses/3JJAY9-8X9/longest_common_prefix.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    var shortestL=A[0].length;
    var tmp=0;
    var size=A.length;
    var LCP="";
    var shortestString=A[0];
    for(var i=1;i<size;i++){
       tmp=A[i].length;
       if(tmp<shortestL){
           shortestL=tmp;
           shortestString=A[i];
       }
   }
   var cc= true;
   for(var i=0;i<shortestL;i++){
       for(var j=0;j<size;j++){
            if(shortestString[i]==A[j][i]){}
            else{
                cc=false;
                break;
            }
       }
       if(cc==false){break;}
       else{
           LCP+=shortestString[i];
       }
   }
   return LCP;
}

File: responses/YJEH95-JPQ/longest_common_prefix.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    if not len(A):
        return ""

    tmp = []
    for item in A[1:]:
        first = A[0]
        while len(first) > 0:
            if first in item:
                tmp.append(first)
                break
            else:
                first = first[:-1]


    if tmp == []:
        return ""
